metrics-jersey/src/main/java/io/stargate/metrics/jersey/MetricsBinder.java

```diff
@@ -0,0 +1,190 @@
+  private final String module;
+  private final MetricsListenerConfig meterListenerConfig;
+  private final MetricsListenerConfig counterListenerConfig;
+
+  /**
+   * Default constructor. Uses {@link SystemPropsMetricsListenerConfig} for metering and counting
+   * configuration.
+   *
+   * @param metrics {@link Metrics} instance.
+   * @param httpMetricsTagProvider Global {@link HttpMetricsTagProvider} registered in the OSGi
+   * @param module Module name
+   */
+  public MetricsBinder(
+      Metrics metrics, HttpMetricsTagProvider httpMetricsTagProvider, String module) {
+    this(
+        metrics,
+        httpMetricsTagProvider,
+        module,
+        new SystemPropsMetricsListenerConfig("stargate.metrics.http_meter_listener"),
+        new SystemPropsMetricsListenerConfig("stargate.metrics.http_counter_listener"));
+  }
+
+  /**
+   * Secondary constructor.
+   *
+   * @param metrics {@link Metrics} instance.
+   * @param httpMetricsTagProvider Global {@link HttpMetricsTagProvider} registered in the OSGi
+   * @param module Module name
+   * @param meterListenerConfig config for metering HTTP requests
+   * @param counterListenerConfig config for counting HTTP requests
+   */
+  public MetricsBinder(
+      Metrics metrics,
+      HttpMetricsTagProvider httpMetricsTagProvider,
+      String module,
+      MetricsListenerConfig meterListenerConfig,
+      MetricsListenerConfig counterListenerConfig) {
+    this.metrics = metrics;
+    this.httpMetricsTagProvider = httpMetricsTagProvider;
+    this.module = module;
+    this.meterListenerConfig = meterListenerConfig;
+    this.counterListenerConfig = counterListenerConfig;
+  }
+
+  /**
+   * Registers application listeners in the {@link JerseyEnvironment}, based on the {@link
+   * #meterListenerConfig} and {@link #counterListenerConfig}.
+   *
+   * @param jersey {@link JerseyEnvironment}
+   */
+  public void register(JerseyEnvironment jersey) {
+    if (meterListenerConfig.isEnabled()) {
+      JerseyTagsProvider meterTagsProvider =
+          getMeterTagsProvider(meterListenerConfig, metrics, httpMetricsTagProvider, module);
+      MetricsApplicationEventListener listener =
+          new MetricsApplicationEventListener(
+              metrics.getMeterRegistry(),
+              meterTagsProvider,
+              StargateMetricConstants.METRIC_HTTP_SERVER_REQUESTS,
+              true);
+      jersey.register(listener);
+    }
+
+    if (counterListenerConfig.isEnabled()) {
+      JerseyTagsProvider counterTagsProvider =
+          getCounterTagsProvider(counterListenerConfig, metrics, httpMetricsTagProvider, module);
+      CounterApplicationEventListener listener =
+          new CounterApplicationEventListener(
+              metrics.getMeterRegistry(),
+              counterTagsProvider,
+              StargateMetricConstants.METRIC_HTTP_SERVER_REQUESTS_COUNTER);
+      jersey.register(listener);
+    }
+  }
+
+  private static JerseyTagsProvider getMeterTagsProvider(
```

Comment:
what's the difference between this and `getCounterTagsProvider`? Can't we have one method for it?